src/EventStore.php

```diff
@@ -359,7 +350,7 @@ public function rollback(): void
 
         $this->actionEventEmitter->dispatch($event);
 
-        $this->recordedEvents = new ArrayIterator();
+        $this->resetRecordedEvents();
```

Comment:
@oqq In a domain model I would do it exactly like you: clean and readable code that tells you what it does. But @sandrokeil 's reasoning wins here, because prooph is infrastructure and performance has very high priority.